src/sentry/tasks/base.py

```diff
@@ -58,6 +65,60 @@ def __call__(self, decorated_task: Any) -> Any:
         return limited_func
 
 
+def taskworker_override(
+    celery_task_attr: Callable[P, R],
+    taskworker_attr: Callable[P, R],
+    namespace: str,
+    task_name: str,
+) -> Callable[P, R]:
+    def override(*args: P.args, **kwargs: P.kwargs) -> R:
+        rollout = 0
+        option_flag = f"taskworker.{namespace}.rollout"
+        if options.isset(option_flag):
+            rollout_map = options.get(option_flag)
+            rollout = rollout_map.get(task_name, 0)
+
+        random.seed(datetime.now().timestamp())
+        if rollout > random.random():
+            metrics.incr(
+                "taskbroker.rollout.shimmed", tags={"namespace": namespace, "task_name": task_name}
+            )
```

Comment:
I'm not sure we need an explicit metric for this. `Tasknamespace.send_task()` also records a metric with the namespace as a tag. We should likely add   `taskname` to that metric's tags as well though.